src/storage/txn/commands/check_secondary_locks.rs

```diff
@@ -49,8 +119,8 @@ impl<S: Snapshot, L: LockManager> WriteCommand<S, L> for CheckSecondaryLocks {
         let mut released_locks = ReleasedLocks::new(self.start_ts, TimeStamp::zero());
         let mut result = SecondaryLocksStatus::Locked(Vec::new());
 
-        for key in self.keys {
-            let (status, released) = txn.check_secondary_lock(&key, self.start_ts)?;
+        for key in mem::take(&mut self.keys) {
+            let (status, released) = self.check_secondary_lock(&mut txn, &key)?;
```

Comment:
By moving the for loop into `check_secondary_lock`, we can avoid passing the key at all. Taking this further, have you tried inlining `check_secondary_lock` here completely? Then it can operate directly on `result` without susing a status. Then we can factor out some smaller functions.

Reply:
> By moving the for loop into `check_secondary_lock`, we can avoid passing the key at all. Taking this further, have you tried inlining `check_secondary_lock` here completely? Then it can operate directly on `result` without susing a status. Then we can factor out some smaller functions.

I thought it might make `process_write` too long, but if it is ok with you, I'll have a try.

Reply:
> I thought it might make process_write too long, but if it is ok with you, I'll have a try.

Too long is fine, we can make it shorter by factoring out functions. I'm pretty sure we'd factor out different ones.